### PR TITLE
CEXT-6071: Add PATCH endpoint for partial config updates

### PR DESCRIPTION
**Pull request by obarcelonap**
## Description
  - Add `PATCH ` handler to the config runtime action accepting partial payloads and null values for unsetting fields
  - Deprecate `PUT ` (still functional, returns `Deprecation: true` header)
  - Widen `sanitizeRequestEntries` to accept nullable config entries
  - Add unit tests covering PATCH semantics, null pass-through, masked-password behaviour, and deprecation header

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have read the **DEVELOPMENT** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/aio-commerce-lib-app/source/actions/config.ts [hunk 1]
```diff
@@ -143,6 +147,38 @@ router.put("/", {
       },
     );
 
+    result.config = filterPasswordFields(configSchema, result.config);
+    return ok({
+      body: result,
+      headers: { "Cache-Control": "no-store", Deprecation: "true" },
```
Comment (iivvaannxx) [inline packages/aio-commerce-lib-app/source/actions/config.ts]: Maybe we should use a custom header syntax? `x-deprecated`?
```suggestion
      headers: { "Cache-Control": "no-store", "x-deprecated": "true" },
```

Comment (obarcelonap) [reply packages/aio-commerce-lib-app/source/actions/config.ts]: deprecation header is standard since RFC9745, adjusted the value to be compliant though.

File: packages/aio-commerce-lib-app/source/actions/config.ts [hunk 2]
```diff
@@ -143,6 +147,41 @@ router.put("/", {
...
+  }),
+
+  handler: async (req, ctx) => {
+    const { logger, rawParams } = ctx;
+    const { configSchema } = rawParams;
+
+    logger.debug(`Patching configuration with scope id: ${req.body.scopeId}`);
+    const { scopeId, config } = req.body;
+
+    const result = await setConfiguration({ config }, byScopeId(scopeId), {
```
Comment (oshmyheliuk) [inline packages/aio-commerce-lib-app/source/actions/config.ts:193]: Does it work without initializing the configuration? Our Docs says that `initialize` must be executed before setting the configuration.

```
You must initialize the library with your configuration schema before using any configuration functions. Calling configuration functions without initialization will throw an error.
```

Comment (obarcelonap) [reply packages/aio-commerce-lib-app/source/actions/config.ts:193]: good catch, it seems the issue was already there because PUT did not called the `initialize` function. Fixed.

Comment (iivvaannxx) [reply packages/aio-commerce-lib-app/source/actions/config.ts:193]: If it didn't call `initialize` and it worked, I think we might need to clarify the docs. Could it be that we only throw in some cases?

File: packages/aio-commerce-lib-config/test/unit/config-utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        name: "currency",
+        value: "USD",
+        origin: { code: "global", level: "global" },
+      },
+      {
+        name: "locale",
+        value: "en_US",
+        origin: { code: "global", level: "global" },
+      },
+    ];
```
Comment (iivvaannxx) [inline packages/aio-commerce-lib-config/test/unit/config-utils.test.ts]: Can we use any of the ones in `fixtures`? If not, maybe add one there?

File: .changeset/yummy-beans-jump.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+"@adobe/aio-commerce-lib-app": minor
+"@adobe/aio-commerce-lib-config": minor
+---
+
+Add PATCH handler to the config runtime action accepting partial payloads and null values for unsetting fields
```
Comment (iivvaannxx) [inline .changeset/yummy-beans-jump.md]: The changeset should describe separate changes. You need two of them. Otherwise you would be creating an update for `lib-config` that says "Add PATCH handler", when `lib-config` doesn't even expose any runtime action.

This PR is modifying two separate packages in two different ways (two Changesets):

- `lib-config`: Adding support for config unsetting
- `lib-app`: Adding that PATCH endpoint and deprecating the `PUT`

**Review by iivvaannxx — APPROVED** [review]

File: .changeset/yummy-beans-jump.md (hunk 4 above)
Comment (obarcelonap) [reply .changeset/yummy-beans-jump.md]: done

File: packages/aio-commerce-lib-config/test/unit/config-utils.test.ts (hunk 3 above)
Comment (obarcelonap) [reply packages/aio-commerce-lib-config/test/unit/config-utils.test.ts]: done

File: .changeset/quiet-dragons-listen.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+"@adobe/aio-commerce-lib-app": minor
+---
+
+Adding new PATCH endpoint for configuration which allows to set configuration values.
+Deprecating PUT endpoint in favor of PATCH.
```
Comment (iivvaannxx) [inline .changeset/quiet-dragons-listen.md]:
```suggestion
Adding new PATCH endpoint for configuration which allows to partially set configuration values. Deprecating PUT endpoint in favor of PATCH.
```

File: packages/aio-commerce-lib-app/source/actions/config.ts [hunk 6]
```diff
@@ -143,6 +153,47 @@ router.put("/", {
...
+    const { configSchema } = rawParams;
+
+    logger.debug(`Patching configuration with scope id: ${req.body.scopeId}`);
+    const { scopeId, config } = req.body;
+
+    const validatedSchema = validateCommerceAppConfigDomain(
+      configSchema,
+      "businessConfig.schema",
+    );
+    initialize({ schema: validatedSchema });
```
Comment (oshmyheliuk) [inline packages/aio-commerce-lib-app/source/actions/config.ts:191]: The PUT endpoint has the logic to filter masked fields
```
    // The UI sent it to us as a masked value, which means the user didn't update it.
    const updatedFields = config.filter(
      (item) => item.value !== MASKED_PASSWORD_VALUE,
    );
```
Should it also be added to the PATCH handler?

Comment (obarcelonap) [reply packages/aio-commerce-lib-app/source/actions/config.ts:191]: My assumption is that since the UI was sending all the config always this was a trick to detect if a masked value was changed.
In the PATCH endpoint this do not happen anymore since the contract has changed to only send what has changed.

Comment (oshmyheliuk) [reply packages/aio-commerce-lib-app/source/actions/config.ts:191]: Anyway, I would keep it just in case, so as not to override passwords with a bunch of asterisks.

**Review by oshmyheliuk — APPROVED** [review]
